ProcessMaker/processmaker
# Implement Search Functionality for My Screen Templates Tab

This PR introduces the search functionality for the My Screen Templates tab. Users can now search for specific screen templates based on their names, descriptions, and category.

## Solution

- Implemented the `ExtendedPMQL` trait for the Screen Templates model to enable advanced search capabilities.
- Developed the `PmqlInput` component to provide a user-friendly interface for entering search queries.
- Added the `index` method for Screen Templates to handle the search functionality and display the results accordingly.

## How to Test

1. Navigate to the My Screen Templates tab.
2. Utilize the search input field to enter a query (e.g., screen template name or description).
3. Verify that the search results update dynamically as you type.
4. Try different search queries to ensure accurate and relevant results are displayed.

## Related Tickets & Packages
- [FOUR-13483](https://processmaker.atlassian.net/browse/FOUR-13483)

ci:next
ci:deploy

## Code Review Checklist


[FOUR-13483]: https://processmaker.atlassian.net/browse/FOUR-13483?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
